plugins/woocommerce-blocks/assets/js/blocks/product-filter/inner-blocks/attribute-filter/edit.tsx

```diff
@@ -138,6 +142,47 @@ const Edit = ( props: EditProps ) => {
 		isEditor: true,
 	} );
 
+	const { productFilterWrapperBlockId, productFilterWrapperHeadingBlockId } =
+		useSelect(
+			( select ) => {
+				if ( ! clientId )
+					return {
+						productFilterWrapperBlockId: undefined,
+						productFilterWrapperHeadingBlockId: undefined,
+					};
+
+				const { getBlockParentsByBlockName, getBlock } =
+					select( 'core/block-editor' );
+
+				const parentBlocksByBlockName = getBlockParentsByBlockName(
+					clientId,
+					'woocommerce/product-filter'
+				);
+
+				if ( parentBlocksByBlockName.length === 0 )
+					return {
+						productFilterWrapperBlockId: undefined,
+						productFilterWrapperHeadingBlockId: undefined,
+					};
+
+				const parentBlockId = parentBlocksByBlockName[ 0 ];
+
+				const parentBlock = getBlock( parentBlockId );
+				const headerGroupBlock = parentBlock?.innerBlocks.find(
+					( block ) => block.name === 'core/group'
+				);
+				const headingBlock = headerGroupBlock?.innerBlocks.find(
+					( block ) => block.name === 'core/heading'
+				);
```

Comment:
If the header group block is wrapped in another group for layout purpose, this is likely to be failed to get the heading. I think we should recursively find the first heading block instead of relying on a fixed structure.

Reply:
It would also fail if someone added another Heading block before the current heading. The only way to always find the correct block would be if the core/heading block accepted some custom attribute or way to identify it as the “default” block heading. Do you know if this is possible?

It’s impossible to cover all possibilities, so we should focus on providing the best experience for the way the design was initially intended. Users are welcome to modify and customize as they want; however, it comes at a cost where some functionalities might not work the same as when the block is used without modifications to the layout structure, etc.

Reply:
> It would also fail if someone added another Heading block before the current heading

I agree that we shouldn't touch user created content, so let's leave it as it is now 👍🏽